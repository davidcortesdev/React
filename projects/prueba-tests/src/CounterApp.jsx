import { useState } from 'react'
import './CounterApp.css'
import PropTypes from 'prop-types'

function CounterApp({contad}) {
  const [count, setCount] = useState(contad)

  const handleSuma = () => setCount(count + 1)
  const handleResta = () => setCount(count - 1)
  const handleReset = () => setCount(contad)

  return (
    <>
      <h1>Contador</h1>
      <h2>{count}</h2>
      <button aria-label="btn-suma" onClick={handleSuma}>+1</button>
      <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
      <button onClick={handleResta}>-1</button>
    </>
  )
}

CounterApp.propTypes = {
  contad: PropTypes.number
}

export default CounterApp

