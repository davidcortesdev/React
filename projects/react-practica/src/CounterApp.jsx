import PropTypes from "prop-types";
import { useState } from "react";

function CounterApp ( {value} ) {

  const [contador, setContador] = useState(value)

  const handleSuma = () => setContador(contador+1)
  const handleResta = () => setContador(contador-1)
  const handleReset = () => setContador(value)

  return (
    <>
    <h1>Contador:</h1>
      <p>{contador}</p>
      <button onClick={(handleSuma)}>Sumar</button>
      <button onClick={handleResta}>Restar</button>
      <button onClick={handleReset}>Restar</button>
    </>
  )
  
}

CounterApp.propTypes = {
  value: PropTypes.number
}

export default CounterApp;

