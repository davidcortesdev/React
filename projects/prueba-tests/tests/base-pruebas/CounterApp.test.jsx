/* eslint-disable no-undef */
import { fireEvent, render, screen } from '@testing-library/react'
import CounterApp from './../../src/CounterApp'

describe('Pruebas en <CounterApp />', () => { 
    
    const initialValue = 100
    const initialValue2 = 100
    
    test('Debe hacer match con el snapshot', () => { 
        
        
        const {container} = render( <CounterApp contad={initialValue}/>)
        expect(container).toMatchSnapshot()
        console.log(initialValue)
     })

     test('Debe inicializar el contador a 100', () => { 
        render( <CounterApp contad={ 100 }/>)
        expect( screen.getByText(100)).toBeTruthy()
        //expect( screen.getByRole('heading', {level: 2}).innerHTML ).toContain('100')
      })

      
      test('Debe de incrementar con el botón +1', () => {   
        render( <CounterApp contad={initialValue} />)
        fireEvent.click( screen.getByText('+1'))
        expect( screen.getByText(101)).toBeTruthy()
       })
       

       test('Debe de decrementar con el botón -1', () => { 
        render( <CounterApp contad={initialValue} />)
        fireEvent.click( screen.getByText('-1'))
        //screen.debug()
        expect( screen.getByText(99)).toBeTruthy()
       })

       test('Debe de resetear a 0 el contador', () => { 
        render( <CounterApp contad={initialValue2} />)
        fireEvent.click( screen.getByRole('button', { name: 'btn-suma' }) )
        fireEvent.click( screen.getByRole('button', { name: 'btn-suma' }) )
        //fireEvent.click( screen.getByText('Reset'))
        //Ponemos aria-label en el boton para que lo coja como name: btn-reset
        fireEvent.click( screen.getByRole('button', { name: 'btn-reset' }) )
        screen.debug()
        expect( screen.getByRole('heading', {level: 2}).innerHTML ).toContain(`${initialValue2}`)
       })

 }) 