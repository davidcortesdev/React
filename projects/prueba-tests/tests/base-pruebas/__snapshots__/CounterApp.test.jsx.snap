// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Contador
  </h1>
  <h2>
    100
  </h2>
  <button
    aria-label="btn-suma"
  >
    +1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
  <button>
    -1
  </button>
</div>
`;
